BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsClient.kt

```diff
@@ -113,12 +113,13 @@ internal class AnalyticsClient(
                         IntegrationType.fromString(integration)
                     )
                     val analyticsRequest = createFPTIPayload(authorization, eventBlobs, metadata)
-                    httpClient.post(
-                        FPTI_ANALYTICS_URL,
-                        analyticsRequest.toString(),
-                        configuration,
-                        authorization
+
+                    val request = BraintreeHttpRequest(
```

Comment:
nit but do we need to call this `BraintreeHttpRequest` or can we just use `HttpRequest`? It feels a bit odd especially here since the URL is technically a PayPal URL, so we aren't actually building a BT request: https://github.com/braintree/braintree_android/blob/9bcdb877cfc22c07ce0128a071a6691f1c76509b/BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsClient.kt#L247

Reply:
So we do already have an `HttpRequest`. We can iterate on the best name, but the main purpose of `BraintreeHttpRequest` is to allow feature clients to make HTTP-like calls without having to know how the requests will be authenticated. `BraintreeGatewayRequest` could work too or something like it.

Reply:
> It feels a bit odd especially here since the URL is technically a PayPal URL, so we aren't actually building a BT request

This is a great callout, but may also be a separate issue. We're still authenticating FPTI calls with BT since those requests used to get routed to Arachne. Does it still work this way on iOS?

Reply:
Maybe we could use `InternalHttpRequest` to capture both PayPal and BT HTTP requests?

> We're still authenticating FPTI calls with BT since those requests used to get routed to Arachne. Does it still work this way on iOS?

We shouldn't need to do this, we don't do anything with auth to the gateway for FPTI requests.

Reply:
Internal could work. The main reason it's `BraintreeHttpRequest` though is because they get sent through `BraintreeHttpClient`. The `BraintreeHttp` prefix is mostly to highlight the compound BT/HTTP domain that the request and the client both belong to.

Also we are still authenticated FPTI calls [on iOS](https://github.com/braintree/braintree_ios/blob/main/Sources/BraintreeCore/Analytics/BTAnalyticsService.swift#L82). I think it's something that we haven't discussed since moving to FPTI. The requirements changed for Analytics as a whole, but on both platforms we still use BT patterns to communicate with that endpoint.

Reply:
I think I am confused by what you mean by "authenticating FPTI". We have logic built around passing in the custom URL which in this case is `https://api.paypal.com/v1/tracking/batch/events`, but we do not hit the BT GW before FPTI, PayPal APIs have their own authentication mechanism separate from Braintree.

On iOS we use 1 pattern for all requests irrespective of BT vs PP APIs when building the request but initialize the `BTHTTP` class in a different way depending on the API we are hitting.

Reply:
Oh ok there's an if statement baked [into BTHTTP here](https://github.com/braintree/braintree_ios/blob/c17642dad8f8a45f1b2d51c0c762873d35e17f6e/Sources/BraintreeCore/BTHTTP.swift#L190). That's my bad I hadn't looked into it deeper.

For us `BraintreeClient` fetches configuration ([similar to APIClient](https://github.com/braintree/braintree_ios/blob/main/Sources/BraintreeCore/BTAPIClient.swift) on iOS). I think we've found a similar piece of technical debt between the two platforms. Does the name for `BTHTTP` make sense if it's also making requests to PayPal endpoints?

I think we can find architectural parity between iOS and Android for this. It might be out of scope for this PR, but it's definitely something we should consider ironing out.

Reply:
Nah no worries, it's complicated for sure. The `BT` prefix only exists because Obj-C requires a 2-3 letter prefix for all classes, but it's not BT specific at all. In the future now that the entire SDK is Swift we can drop the prefix and add a specific annotation just for the Obj-C extension, but it does not necessary denote BT specifics, just a requirement for Obj-C interoperability.

Agree there is some code smell that we can handle in a separate PR. For this PR maybe we can run with `Internal` for now?

Reply:
Yeah that makes sense for now. I'm down to go with Internal.